haystack/schema.py

```diff
@@ -94,17 +96,52 @@ def __str__(self):
         return str(self.to_dict())
 
 
+@dataclass
+class Answer:
+    answer: Optional[str]
+    type: Literal["generative", "extractive"]
+    #question: Optional[str]
+    score: Optional[float]
+    #probability: Optional[float] = None
+    context: Optional[str]
+    offset_start: Optional[int]
+    offset_end: Optional[int]
+    offset_start_in_doc: Optional[int]
+    offset_end_in_doc: Optional[int]
+    document_id: Optional[str]
```

Comment:
Just curious which case Answer will have `document_id`?

Reply:
In almost every case. It's basically the document where the answer was found. This link is quite important as it allows in a GUI to show the "original document" related to the search result. It's also needed for storing user feedback about search results and creating continuous training data.

Reply:
I feel these all (document id, chunk id, file name etc) should be in meta. But align with you :)